packages/java/src/node/java-contribution.ts

```diff
@@ -48,14 +50,14 @@ export class JavaContribution extends BaseLanguageServerContribution {
         protected readonly contributions: ContributionProvider<JavaExtensionContribution>
     ) {
         super();
+        this.connections = new Map<IConnection, string>();
         this.javaBundles = [];
         for (const contrib of this.contributions.getContributions()) {
             this.javaBundles = this.javaBundles.concat(contrib.getExtensionBundles());
         }
     }
 
-    start(clientConnection: IConnection): void {
-
+    async start(clientConnection: IConnection, parameters?: any): Promise<void> {
```

Comment:
I'd prefer `object` instead of `any`.

@akosyakov, WDYT? would it be okay to pass params from client contribution like this? it can be useful in other cases, e.g. configuring python ls.

Reply:
I don't mind about adding parameters. I don't understand why this change should go all way down to web socket channels. The open request is supposed to open a web socket channel, nothing more. After the WS-channel has been opened, any protocol could be used over it to pass protocol-specific parameters.

Reply:
Would you rather suggest to delay the LS startup?

Currently the LS integration (I know about) starting the processes on `start` of the BE contribution.

Reply:
I agree that we don't really want to clutter the connection establishment with parameters that are not related to websockets.
The connection that we are opening is destined for being used as a language server communication channel. I think we should NOT use it to run meta-protocol (like "startLanguageServer").  Therefore, I think we should use a general ws channel to start the language server. So the algorithm would become:
1. Establish a ws connection to /languages/<contribution id> the connection to the back end contribution is established, but nothing is sent over it yet. The languages backend contribution remembers the established connection.
2. send "start(parameters)" over a ws channel at /languages, the languages backend contrib now calls "start(connection, parameters)" on the language server contribution. What do you think?

Reply:
@tsmaeder I think the algorithm I described is similar to your. Just one remark: in step 2 we need to send start(**contributionPath**, parameters) to know which ls should be started

Reply:
Changing the languages protocol is fine with me if it does not break APIs and behavior for existing extensions. We have many already outside of the theia repo.